packages/plans-grid-next/src/components/comparison-grid/index.tsx

```diff
@@ -963,46 +964,27 @@ const ComparisonGrid = ( {
 		? getWooExpressFeaturesGrouped()
 		: getPlanFeaturesGrouped();
 
-	let largeBreakpoint;
-	let mediumBreakpoint;
-	let smallBreakpoint;
-
-	if ( isInSignup ) {
-		// Breakpoints without admin sidebar
-		largeBreakpoint = 1281;
-		mediumBreakpoint = 1024;
-		smallBreakpoint = 880;
-	} else {
-		// Breakpoints with admin sidebar
-		largeBreakpoint = 1553; // 1500px + 272px (sidebar)
-		mediumBreakpoint = 1296; // 1340px + 272px (sidebar)
-		smallBreakpoint = 1152; // keeping original breakpoint to match Plan Grid
-	}
-
-	const isLargeBreakpoint = usePricingBreakpoint( largeBreakpoint );
-	const isMediumBreakpoint = usePricingBreakpoint( mediumBreakpoint );
-	const isSmallBreakpoint = usePricingBreakpoint( smallBreakpoint );
-
 	const [ visiblePlans, setVisiblePlans ] = useState< PlanSlug[] >( [] );
 
 	const displayedGridPlans = useMemo( () => {
-		return sortPlans( gridPlans, currentSitePlanSlug, isMediumBreakpoint );
-	}, [ gridPlans, currentSitePlanSlug, isMediumBreakpoint ] );
+		return sortPlans( gridPlans, currentSitePlanSlug, 'small' === gridSize );
+	}, [ gridPlans, currentSitePlanSlug, gridSize ] );
 
 	useEffect( () => {
 		let newVisiblePlans = displayedGridPlans.map( ( { planSlug } ) => planSlug );
 		let visibleLength = newVisiblePlans.length;
 
-		visibleLength = isLargeBreakpoint ? 4 : visibleLength;
-		visibleLength = isMediumBreakpoint ? 3 : visibleLength;
-		visibleLength = isSmallBreakpoint ? 2 : visibleLength;
+		visibleLength = 'large' === gridSize ? 4 : visibleLength;
+		visibleLength = 'medium' === gridSize ? 3 : visibleLength;
+		visibleLength = 'smedium' === gridSize ? 2 : visibleLength;
+		visibleLength = 'small' === gridSize ? 2 : visibleLength;
```

Comment:
These are all being evaluated. Should we instead just use a switch?

```
switch ( gridSize ) {
	case 'large':
		visibleLength = 4;
		break;
	case 'medium':
		visibleLength = 3;
		break;
	...
}
```

I think we should narrow the type for `gridSize` too, using 'large' | 'medium' etc. It's also specified as being potentially undefined - makes we wonder if that's actually the case, or whether it will always have a value 🤔

Reply:
> These lines will all be evaluated. Perhaps simpler as a switch?

Absolutely.

> Note: I think we should narrow the type for gridSize too ie. as 'large' | 'medium' etc. 

I do intend to narrow the type at some point. I'm not entirely sure yet where/how (it will probably depend on whether the flags end up being applied on the `plans-grid-next` class. - as done here - or the individual grid ones: `plans-grid-next__features-grid`, `plans-grid-next__comparison-grid`). From there, the type itself might be worked out differently. I think by the end of next week we'll have the final form.

> It's also specified as being possibly undefined - is that actually the case I wonder 🤔 

The intention is for it to also be an optional prop being passed from the consuming end e.g. one to be able to pass `gridSize: small` and get the mobile version back. I'm not entirely sure yet though, how that will work, but having it as optional seems like the way forward for now.

Reply:
> These lines will all be evaluated. Perhaps simpler as a switch?

Updated to use `switch`. Good call on the process/evaluation :-)